### PR TITLE
Kubernetes HA etcd

### PR DESCRIPTION
**Pull request by LLParse**
Need the end user to specify HOST_FAILURE_TOLERANCE and then compute scale based on that

### CODE REVIEW

**Comment by alena1108** [comment]
Just to track the issue here: https://github.com/rancher/rancher/issues/4499

File: resources/content/config-content/system-stacks/kubernetes/docker-compose.yml [hunk 1]
```diff
@@ -42,26 +42,37 @@ proxy:
...
 
 etcd:
-    command:
-        - /usr/local/bin/etcd
-        - --advertise-client-urls=http://etcd:4001
-        - --listen-client-urls=http://0.0.0.0:4001
-        - --data-dir=/var/etcd/data
-    image: rancher/etcd:2.0.12
+    image: rancher/etcd:v2.3.3
+    command: node 3
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/kubernetes/docker-compose.yml]: floor(HOST_FAILURE_TOLERANCE * 2) + 1

File: resources/content/config-content/system-stacks/kubernetes/docker-compose.yml [hunk 2]
```diff
@@ -42,26 +42,37 @@ proxy:
...
     labels:
         io.rancher.sidekicks: etcd-data
+        io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/kubernetes/docker-compose.yml]: soft -> hard constraint

File: resources/content/config-content/system-stacks/kubernetes/docker-compose.yml [hunk 3]
```diff
@@ -42,26 +42,37 @@ proxy:
...
+etcd-discovery:
+    image: rancher/etcd:v2.3.3
+    command: discovery_node
+    labels:
+        io.rancher.container.start_once: 'true'
+        io.rancher.sidekicks: etcd-bootstrap
+
+etcd-bootstrap:
+    image: rancher/etcd:v2.3.3
+    command: bootstrap 3
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/kubernetes/docker-compose.yml]: floor(HOST_FAILURE_TOLERANCE * 2) + 1

**Comment by cloudnautique** [comment]
Lets look at the `etcd` init script before merging this.

**Comment by alena1108** [comment]
@ibuildthecloud @LLParse @cloudnautique as we are moving kubernetes template from cattle to rancher-catalog ([k8s catalog item PR](https://github.com/rancher/rancher-catalog/pull/150)), we might need to move this code there as well.

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/4341

**Comment by LLParse** [comment]
Closed in favor of https://github.com/rancher/rancher-catalog/pull/156

**Pull request closed without merging**